config.go

```diff
@@ -70,32 +70,63 @@ func (c LoggingConfig) extractor() observability.ContextExtractor {
 }
 
 // MetricsConfig describes how telemetry should be configured.
+// Scope and Tally are exclusive; choose one.
+// Only if Tally is present, the dispatcher will push metrics to Tally.
+// Otherwise, it is the owner of the dispatcher's repsonsibility to publish
+// metrics.
+// Configuring neither a net/metrics.Scope nor a tally.Scope will leave metrics
+// disabled, making all instrumentation no-ops.
 type MetricsConfig struct {
+	// Metrics is a *"go.uber.org/net/metrics".Scope for recording stats.
+	// YARPC does not push these metrics; pushing metrics from the root is an
+	// external concern.
+	Metrics *metrics.Scope
 	// Tally scope used for pushing to M3 or StatsD-based systems. By
 	// default, metrics are collected in memory but not pushed.
 	Tally tally.Scope
 }
 
-func (c MetricsConfig) registry(name string, logger *zap.Logger) (*pally.Registry, context.CancelFunc) {
-	r := pally.NewRegistry(
-		pally.Labeled(pally.Labels{
-			"component":  _packageName,
-			"dispatcher": pally.ScrubLabelValue(name),
-		}),
-		// Also expose all YARPC metrics via the default Prometheus registry.
-		pally.Federated(prometheus.DefaultRegisterer),
-	)
+func (c MetricsConfig) scope(name string, logger *zap.Logger) (*metrics.Scope, context.CancelFunc) {
+	var root *metrics.Root
+	var parent *metrics.Scope
+
+	// c.Metrics, c.Tally
+	// yes, yes, warn, use metrics as parent, do not push to tally
+	// yes, nil, use as parent, do not push to tally
+	// nil, yes, create root, push to tally
+	// nil, nil, use nil scope, do not push to tally
+
+	if c.Metrics != nil {
+		if c.Tally != nil {
+			logger.Warn("yarpc.NewDispatcher expects only one of Metrics.Tally or Metrics.Scope. " +
+				"To push to Tally, either use a Metrics.Scope and use tallypush, or just pass a Tally Scope")
+		}
+		parent = c.Metrics
+	} else {
+		if c.Tally != nil {
+			root = metrics.New()
+			parent = root.Scope()
+		} else {
+			// Neither Metrics nor Tally, so no-op metrics
+			return nil, func() {}
+		}
+	}
```

Comment:
I *think* that this may end up easier to grok if you pick off some of the cases up front. E.g.,

```
if c.Metrics == nil && c.Tally == nil {
  return nil, func() {}
}
...more stuff...
```

Up to your discretion here.